python/ray/actor.py

```diff
@@ -92,9 +92,6 @@ def annotate_method(method):
 class ActorMethod:
     """A class used to invoke an actor method.
 
-    Note: This class only keeps a weak ref to the actor, unless it has been
```

Comment:
This isn't really an acceptable regression. Actor GC delay is much worse.

Reply:
for my understanding, i thought here is the difference before and after:

before:
```
a = Actor.remote()  # a.ref_count = 1
del a.  # a.ref_count = 0 -> actor handle GCable.
```

after:

```
a = Actor.remote()  # a.ref_count = 1 + num(actor_methods)
del a  # a.ref_count = 0 -> actor handle GCable.
```

so basically the problem is the actor_methods GC could be scattered and delay the GC of the actor.

Reply:
Yes, the issue is the circular reference between the actor handle and the actor. It seems possible to fix this, but I'm not sure this PR is the way.

Reply:
this method is always short right? So it is highly likely no impact? Or @ericl are you aware of any other issue?